### ISSUE TITLE
WCAG 2.2: Explore how to handle focus state of short links

### ISSUE THREAD

**Issue by owenatgov** [alphagov/govuk-frontend#4470]
## What
Figure out how to present focus states for links of 4 or fewer characters and for components which use the link styles.

## Why
During testing of components against [WCAG 2.2 criterion 14.2.13: focus appearance](https://www.w3.org/WAI/WCAG22/Understanding/focus-appearance.html) as part of https://github.com/alphagov/govuk-frontend/issues/4347, it was discovered that we have several failures within components to do with parts of these components using our link decoration. The root problem is that that very short links have too small a focus area to pass the criteria. The link style itself _does_ pass the criterion, however components that use the link decoration but not the default link colour eg: breadcrumbs using black links fail this because they lose the contrast between the blue and black that links typically have.

This impacts the following components:

- [Back links](https://design-system.service.gov.uk/components/back-link/)
- [Breadcrumbs](https://design-system.service.gov.uk/components/breadcrumbs/)
- [Error summary](https://design-system.service.gov.uk/components/error-summary/)
- [Pagination](https://design-system.service.gov.uk/components/pagination/)
- [Summary list](https://design-system.service.gov.uk/components/summary-list/) (summary cards fail but summary lists pass)
- [Tabs](https://design-system.service.gov.uk/components/tabs/)

As part of this work, we have an opportunity to also update our [link guidance](https://design-system.service.gov.uk/styles/links/) to encourage users to not use short links in their content.

### Further reading
- [Testing spreadsheet (internal only)](https://docs.google.com/spreadsheets/d/1PpKvhCGZJb1j0ZUUY2ZuA0CBQeu04yFD1bumiL8wGX8/edit#gid=0)
- https://github.com/alphagov/govuk-design-system/issues/3007 - related issue on focus state rationalisation
 
## Who needs to work on this
Developers, designers

## Who needs to review this
Developers, designers

## Done when

- [ ] Solution to how to present short links is devised and merged into `main`
- [ ] Link guidance is updated to advise not using short links
